test/Groups-conformance-tests.jl

```diff
@@ -0,0 +1,298 @@
+                # TODO: throw something more specific
+                @test_throws ErrorException gens(G)
+                @test_throws ErrorException ngens(G)
+            end
+        end
+
+        @testset "order, rand" begin
+            if isfinite(G)
+                @test order(Int16, G) isa Int16
+                @test order(BigInt, G) isa BigInt
+                @test order(G) >= 1
+                @test is_trivial(G) == (order(G) == 1)
+            else
+                @test_throws AbstractAlgebra.InfiniteOrderError order(G)
+                @test !is_trivial(G)
+            end
+
+            @test rand(G) isa GroupElem
+            @test rand(G, 2) isa AbstractVector{elem_type(G)}
+            g, h = rand(G, 2)
+            @test parent(g) === parent(h) === G
+
+#            @test GroupsCore.rand_pseudo(G) isa elem_type(G)
+#            @test GroupsCore.rand_pseudo(G, 2, 2) isa AbstractMatrix{elem_type(G)}
+#
+#            g, h = GroupsCore.rand_pseudo(G, 2)
+#            @test parent(g) === parent(h) === G
+        end
+    end
+end
+
+function test_GroupElem_interface(g::GEl, h::GEl) where {GEl<:GroupElem}
+
+    @testset "GroupElem interface" begin
+
+        @testset "Parent methods" begin
+            @test parent(g) isa Group
+            @test parent(g) === parent(h)
+            G = parent(g)
+
+            @test elem_type(G) == typeof(g)
+
+            @test one(g) isa elem_type(G)
+
+            @test one(G) == one(g) == one(h)
+
+            @test isone(one(G))
+        end
+
+        @testset "Equality, deepcopy && hash" begin
+            @test (g == h) isa Bool
+            @test isequal(g, h) isa Bool
+
+            @test g == g
+            @test isequal(g, g)
+
+            if g != h
+                @test !isequal(g, h)
+            end
+
+            @test deepcopy(g) isa typeof(g)
+            @test deepcopy(g) == g
+            k = deepcopy(g)
+            @test parent(k) === parent(g)
+            @test hash(g) isa UInt
+            @test hash(g) == hash(k)
+
+            if isequal(g, h)
+                @test hash(g) == hash(h)
+            end
+        end
+
+        @testset "Group operations" begin
+            old_g, old_h = deepcopy(g), deepcopy(h)
+
+            # check that the default operations don't mutate their arguments
+            @test inv(g) isa typeof(g)
+            @test (g, h) == (old_g, old_h)
+
+            @test g * h isa typeof(g)
+            @test (g, h) == (old_g, old_h)
+
+            @test g^2 == g * g
+            @test (g, h) == (old_g, old_h)
+
+            @test g^-3 == inv(g) * inv(g) * inv(g)
+            @test (g, h) == (old_g, old_h)
+
+            @test (g * h)^-1 == inv(h) * inv(g)
+            @test (g, h) == (old_g, old_h)
+
+            @test conj(g, h) == inv(h) * g * h
+            @test (g, h) == (old_g, old_h)
+
+            @test ^(g, h) == inv(h) * g * h
+            @test (g, h) == (old_g, old_h)
+
+            @test commutator(g, h) == g^-1 * h^-1 * g * h
```

Comment:
In GroupsCore, we had `commutator`, which we also imported; but we didn't use this anywhere. This is in line with our agreement to have "self-explanatory" names in general (though we sometimes make exceptions)

In Oscar we have `comm`, which is also used there. This is kinda in line with `inv` and `conj` and `gens` ... I think I am going to use that here for now, and leave the decision on whether to rename some or all of these to another time